main.go

```diff
@@ -133,6 +134,34 @@ func main() {
 		config.Output = defaultOutputFilename
 	}
 
+	// Check if file already exists
+	if _, err := os.Stat(config.Output); err == nil {
+		var confirm bool
+		var newOutputFilename string
+
+		confirmOverwriteForm :=
+			huh.NewConfirm().
+				Title(fmt.Sprintf("'%s' already exists. Would you like to overwrite this file?", config.Output)).
+				Value(&confirm)
```

Comment:
should check for if is a directory or not first. If directory, might be better to fail immediately. to save people from themselves.

Reply:
Fair, I shall get to this in like the next 6 hours or something

Reply:
Hey @debdutdeb 

![image](https://github.com/charmbracelet/freeze/assets/42164574/d342c010-08af-4c5b-acb3-4e53f349bc4a)

So I tried the following:
- `freeze.png` already exist
- Ran the following command below
- Entered the new file name as `non-exist/file.png` indicating an example of the directory not existing, and adding a file within a non existent directory
And it failed which is good
```
go run ./... --execute "cat SECURITY.md"
                                      
   ERROR  Unable to convert SVG to PNG
                                      
  open non-exist/file.png: no such file or directory
                                                    
exit status 1
```

Obviously when I added it as a parameter for the output, it fails as expected since the file is being added within a non existent directory, which I think is intended
```
go run ./... --execute "cat SECURITY.md" --output "non-exist/file.png"
                                      
   ERROR  Unable to convert SVG to PNG
                                      
  open non-exist/file.png: no such file or directory
                                                    
exit status 1
```

And finally, when the form pops up and add the filename as `non-exist-dir/non-exist-dir2` to also indicate 2 non existent directory, it fails as expected (yeah even though the `non-exist-dir2` would end up becoming a file atleast with me being on Pop OS)
```
go run ./... --execute "cat SECURITY.md"
                                
   ERROR  Unable to write output
                                
  open non-exist-dir/non-exist-dir2: no such file or directory
                                                              
exit status 1
```

but yeah should I still go ahead and check if the given path is a directory? To me it seems like it ends up handling it, open to whatever!

Reply:
It does but ig that depends on how file is being written and expection for actions after the confirmation. Like code that writes to the files consider them simply files, and might someday want to remove before writing, which would remove the directory not knowing it is a directory.

So I'd suggest still adding that check.

Reply:
Sure thing, will look into it within like 24 hours or something!

Reply:
https://github.com/charmbracelet/freeze/pull/86#issuecomment-2090390765

Gone ahead and resolved it here, lemme know if there's anything else!